File: backend/dataset/coinChange/coinChange_8.py
class Solution:
    def coinChange(self, coins: List[int], amount: int) -> int:
        
        memo = [float("inf")]*(amount+1)
        memo[0] = 0
        
        for i in range(1,amount+1):
            for coin in coins:
                
                if i-coin >= 0 and memo[i] > memo[i-coin]+1:
                    memo[i] = memo[i-coin]+1

        if memo[amount] == float("inf"):
            return -1
        return memo[amount]
		           

File: backend/CDataset/coinChange/coinChange_8.cpp
class Solution {
public:    int coinChange(vector<int>& coins, int amount) {
        vector<int> dp(amount+1, amount+1);
        dp[0] = 0;        
        for(int i = 0; i <= amount; i++){            
            for(auto c : coins){
                if(i - c < 0) {
                    continue;
                }                
                dp[i] = min(dp[i], dp[i-c]+1);
            }            
        }        
        return dp[amount] < amount+1 ? dp[amount] : -1;
    }
};